pkg/operator/ceph/cluster/cluster.go

```diff
@@ -230,12 +234,84 @@ func (c *ClusterController) initializeCluster(cluster *cluster) error {
 	cluster.mons.ClusterInfo = cluster.ClusterInfo
 	cluster.mons.ClusterInfo.SetName(c.namespacedName.Name)
 
+	// Skip the new CSI-operator creation when holder pod is enabled until multus support is added in the CSI operator
+	if csi.EnableCSIOperator && !csi.IsHolderEnabled() {
```

Comment:
In the mon reconcile there is code that updates the mons for the csi driver. I'd suggest we configure the CephConnections CR at that point. See [here](https://github.com/rook/rook/blob/54a294425274922b51abfb539059a4e7b248ec2c/pkg/operator/ceph/cluster/mon/mon.go#L1102-L1113) for internal clusters. See [here](https://github.com/rook/rook/blob/54a294425274922b51abfb539059a4e7b248ec2c/pkg/operator/ceph/cluster/cluster_external.go#L126-L139) for external clusters. We should probably refactor that to a single helper method though if we can avoid the duplication.